Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -172,3 +188,95 @@ function IsFilePathSpecified
                 [StringComparison]::OrdinalIgnoreCase)
 }
 
+
+function GetVersion($uri, $headers)
+{
+    $version = $null
+
+    Try
+    {
+        $jsonResp = Invoke-RestMethod -Uri $uri -Method Get -Headers $headers
```

Comment:
Wondering if this call is made here for first time? if it has been already made at some other place in flow, probably we should just propogate that value here. It may not be that big issue for on-prem but for hosted it matters :).